docs/contributing/contributing-rules.md

```diff
@@ -1,151 +1,376 @@
 
 You can also publish rules to the Semgrep Registry as private rules. See the [Private rules](/docs/writing-rules/private-rules.md) documentation for more information.
 
-### Contributing through GitHub
+## Contributing through GitHub
 
-Fork our repository and make a pull request; we'll contact you about signing our Contributor License Agreement (CLA). Install pre-commit (see [installing pre-commit](#installing-pre-commit)) and make a pull request to the [rules repository](https://github.com/returntocorp/semgrep-rules) with two files:
+Fork our repository and make a pull request; we'll contact you about signing our Contributor License Agreement (CLA). Install pre-commit (see [installing pre-commit](#installing-pre-commit)) and make a pull request to the [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) with two files:
 1. The semgrep pattern (as YAML file).
 2. The test file (with the file extension of the language or framework). The test file must contain at least one true positive and one true negative test case to be approved. See the [Tests](#tests) section of this document for more information.
 
-See an example of a [pull request](https://github.com/returntocorp/semgrep-rules/pull/1728/files) to the rules repository. Pull requests require the approval of at least one maintainer and successfully passed [CI jobs](https://github.com/returntocorp/semgrep-rules/actions).
+See an example of a [pull request](https://github.com/returntocorp/semgrep-rules/pull/1728/files) to the Semgrep Registry. Pull requests require the approval of at least one maintainer and successfully passed [CI jobs](https://github.com/returntocorp/semgrep-rules/actions).
+
+Find more about the Semgrep Registry by reading the [Rule writing](#rule-writing) and [Tests](#tests) sections.
 
-Find more about the rules repository by reading the [Rule writing](#rule-writing) and [Tests](#tests) sections.
+## Rule requirements
 
-#### Installing `pre-commit`
+Every rule submitted to Semgrep Registry must include:
 
-If you are contributing to the rules repository through GitHub but you are **not** contributing through sharing your rule in Semgrep App, install `pre-commit`. Follow [pre-commit installation documentation](https://pre-commit.com/#installation). Once `pre-commit` is set up you may commit code and create pull requests to rules repository.
+- Rule ID - `ruleid`. Rule ID is created automatically by Semgrep App, but include it if you are contributing solely through GitHub without Semgrep App.
+- Test file. See [Tests](#tests) for more details. Test file must include:
+    - Code that Semgrep detects using the rule.
+    - Code that is clearly shown as `ok` and Semgrep should not detect it.
+- Rule message, for more information see [Rule messages](#rule-messages).
+- Severity TODO
```

Comment:
@LewisArdern as discussed, please fill in the required fields for Semgrep Registry rules in general (Technology, Severity, not sure what all are we missing). I will convert this list to a table later.

Also documenting optional fields might be a task for another PR...

Let me know if this list is ok for now, we should also provide a rule ID definition - can it be: `Rule ID is a unique identifier of a rule in Semgrep Registry.` How to create a unique rule ID when contributing only through GitHub?

Reply:
It shouldn't matter what people call their rule, because the way the registry produces rules is based on 

filename.yaml -> id: test
filename1.yaml -> id:test

These would be

filename1.test
filename.test 

^ which would be unique in the registry. 

So in general they just need to write a meaningful name for the ID as it stands today.

Reply:
I updated the technology and reference, that should cover most of it.